### PR TITLE
Oe/86bz12uch/login stepper v2

### PR DESCRIPTION
**Pull request by guardian-001**
This PR presents the second version of the login stepper for suppliers
## Note !! : 
- I ran this PR just to ensure that the number of files changed is not high. 
- The work is not complete
- folders and files structure will be updated in the last pr

### CODE REVIEW

File: src/modules/supplier/create-account/hooks/use-details.ts
Comment (MohamedMegdich1) [inline src/modules/supplier/create-account/hooks/use-details.ts:1]: use-details you should change the naming

File: src/app/(supplier)/(private)/(complete)/complete-account.tsx
Comment (MohamedMegdich1) [inline src/app/(supplier)/(private)/(complete)/complete-account.tsx:1]: remove unused it code

File: src/api/common/client.tsx [hunk 1]
```diff
@@ -72,3 +72,12 @@ client.interceptors.response.use(
     return Promise.reject(error);
   }
 );
+
+export const post = async (url: string, data: any) => {
+  return client
```
Comment (MohamedMegdich1) [inline src/api/common/client.tsx:77]: handle the error and success on the mutation

File: src/services/shared/auth-services.ts [hunk 2]
```diff
@@ -1,24 +1,28 @@
...
 
 export async function postSignup(request: SignupRequest): Promise<Response> {
   const url = '/api/architect-request/create-architect-request/';
-  return client({
-    url: url,
-    method: 'POST',
-    data: request,
-  })
+  return client
+    .post(url, request)
```
Comment (MohamedMegdich1) [inline src/services/shared/auth-services.ts:16]: handle the error and success on the mutation

File: src/services/shared/supplier-services.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { isAxiosError } from 'axios';
+
+import { client } from '@/api';
+import type { SpecialityTypeResponse } from '@/api/auth';
+
+export async function getSpecialityTypes(): Promise<SpecialityTypeResponse[]> {
+  const url = 'api/users/supplier/speciality-types/';
+  return client
+    .get(url)
```
Comment (MohamedMegdich1) [inline src/services/shared/supplier-services.ts:9]: handle the error and success on the mutation

**Review by MohamedMegdich1 — CHANGES_REQUESTED** [review]

File: src/modules/supplier/create-account/hooks/use-details.ts
Comment (guardian-001) [reply src/modules/supplier/create-account/hooks/use-details.ts:1]: done
use-company-details.ts

File: src/app/(supplier)/(private)/(complete)/complete-account.tsx
Comment (guardian-001) [reply src/app/(supplier)/(private)/(complete)/complete-account.tsx:1]: uncommented

File: src/api/common/client.tsx (hunk 1 above)
Comment (guardian-001) [reply src/api/common/client.tsx:77]: function removed, there is default post method in axios

File: src/services/shared/auth-services.ts (hunk 2 above)
Comment (guardian-001) [reply src/services/shared/auth-services.ts:16]: i fixed it by handling the error out of the service like the others : 
image
image
image

File: src/services/shared/supplier-services.ts (hunk 3 above)
Comment (guardian-001) [reply src/services/shared/supplier-services.ts:9]: it's a fetcher not mutation

**Pull request merged**